libratbag: init at v0.9.903

mvnetbiz opened the pull request:
###### Motivation for this change
Libratbag is a daemon and utility for configuring DPI, buttons, etc. of programmable gaming mice.

###### Things done
Added package for libratbag and service module ratbagd.
Added piper: gtk frontend app for ratbagctl
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

jtojnar commented on nixos/modules/services/hardware/ratbagd.nix:
```diff
@@ -0,0 +1,40 @@
+    # Give users access to the "ratbagctl" tool
+    environment.systemPackages = [ pkgs.libratbag ];
+
+    systemd.services.ratbagd = {
```
You should be able to use `systemd.packages = [ pkgs.libratbag ];`

jtojnar commented on pkgs/os-specific/linux/libratbag/default.nix:
```diff
@@ -0,0 +1,32 @@
+  buildInputs = [ glib systemd udev libevdev gitMinimal check valgrind swig ];
+
+
+  enableParallelBuilding = true;
```
`meson` builds in parallel by default.

jtojnar commented:
See also https://github.com/NixOS/nixpkgs/issues/35516

mvnetbiz commented on pkgs/os-specific/linux/libratbag/default.nix:
```diff
@@ -0,0 +1,32 @@
+  };
+
+  nativeBuildInputs = [
+    (python3.withPackages (ps: with ps; [ evdev pygobject3 ]))
```
If this is moved to buildInputs, the patch shebangs for "ratbagctl" gives the wrong python, and program fails because it can't import the right modules. This seemed like the wrong place to put it based on nixos manual's description of the different build inputs, but I am not sure how to get around that.

mvnetbiz commented:
I've included your suggestions, thanks.

jtojnar commented on nixos/modules/services/hardware/ratbagd.nix:
```diff
@@ -0,0 +1,41 @@
+    environment.systemPackages = [ pkgs.libratbag ];
+
+    systemd.packages = [ pkgs.libratbag ];
+    systemd.services.ratbagd = {
```
Why did you leave this? It should already be specified in the service file in the package, which is installed by the `systemd.packages` option.

mvnetbiz replied:
I couldn't figure out of that actually used the packages own unit file. It didn't seem to work for me when I removed my own definition. Is there a way to point it to where the build outputs it?

jtojnar replied:
It does, only the servcie file does not have `wantedBy`. You can do `systemd.services.ratbagd.wantedBy = [ "multi-user.target" ];`, which will add the key to the [override](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#id-1.14.3).

I wonder why the upstream service does not have that, though – maybe the daemon is supposed to be launched using a different method.

jtojnar commented:
Do you plan to package piper as well? It would be nice to have some app to test if this is working.

mvnetbiz commented:
I was going to do that too. Would that be a separate PR? Libratbag package does have a command line that I have tested with my Logitech G403.

jtojnar commented:
For closely related packages like this, a single PR is fine.

jtojnar commented:
I am aware of the issue and it will be fixed in the next meson release (#46020). At the moment we try to fix this by adding `python` before `meson` in `nativeBuildInputs` but I am not sure 100% sure it will work for Python programs. What expression do you have?

mvnetbiz commented:
Pushed my expression for piper

mvnetbiz commented:
I patched out 2 lines from meson.build. One is it can't verify python gobject is installed. This must have something to do with pkg-config? I don't know how any of that works. And the other line includes a script that does something with gtk-update-icon-cache which fails. I think we don't need icon cache in the store anyways?

jtojnar commented:
Yep, only packages listed in build inputs get added to the `PKG_CONFIG_PATH`. You can add pygobject there.

Right, `gtk-update-icon-cache` is not good for us, we have `hicolor-icon-theme` setup hook removing the generated cache file for us, though, maybe we should still run the file in case the glib-compile-schemas is ever uncommented. See, for example, this for how to make it runnable:

https://github.com/NixOS/nixpkgs/blob/874b045cc891e45f7e391a9fca16aa69b45320ff/pkgs/desktops/gnome-3/apps/gnome-calendar/default.nix#L29-L32

mvnetbiz commented:
Ok I've fixed up piper, using the python builder now. It works with my logitech mouse.

mvnetbiz commented:
Is there anything else I need to do? Should I squash commits or anything?

jtojnar commented:
Each package should be in separate commit, otherwise it looks good.

mvnetbiz committed:
libratbag: init at v0.9.903
Add package libratbag and service module ratbagd
Libratbag contains ratbagd daemon and ratbagctl cli to configure
buttons, dpi, leds, etc. of gaming mice.
Add mvnetbiz to maintainers.

mvnetbiz committed:
piper: init at 0.2.902
Add package piper. Piper is a gtk frontend application for ratbagctl
to configure gaming mice.

jtojnar commented:
Thank you.
